### PR TITLE
Test coverage rc6

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/JCommander.kt [hunk 1]
```diff
@@ -1,8 +1,7 @@
-package io.gitlab.arturbosch.detekt.cli
+package io.gitlab.arturbosch.detekt.generator
```
Comment (vanniktech) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/JCommander.kt]: Would be nice to have a detekt rule for this wouldn't it?

Comment (schalkms) [reply detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/JCommander.kt]: Good idea! Would you mind to create an issue for this feature?

Comment (vanniktech) [reply detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/JCommander.kt]: There you go #646

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/DuplicateCaseInWhenExpression.kt [hunk 2]
```diff
@@ -48,12 +48,22 @@ class DuplicateCaseInWhenExpression(config: Config) : Rule(config) {
...
-		val distinctEntries = entries.distinct()
+		val duplicates = findDuplicates(entries)
+		if (duplicates.isNotEmpty()) {
+			report(CodeSmell(issue, Entity.from(expression),
+					"When expression has multiple case statements " + "for ${duplicates.joinToString { ", " }}."))
+		}
+	}
 
-		if (entries.size > distinctEntries.size) {
-			val duplicateExpressions = entries - distinctEntries
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/DuplicateCaseInWhenExpression.kt]: do you have a test case where this was not enough? just for the test coverage :)

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/DuplicateCaseInWhenExpression.kt]: This fixes a bug. I have described the cause in the first commit. This has nothing to do with the coverage.

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/DuplicateCaseInWhenExpression.kt]: Ah ok!

**Pull request merged**